engine/storage/snapshot/src/org/apache/cloudstack/storage/snapshot/XenserverSnapshotStrategy.java

```diff
@@ -279,7 +282,52 @@ public boolean deleteSnapshot(Long snapshotId) {
 
     @Override
     public boolean revertSnapshot(SnapshotInfo snapshot) {
-        throw new CloudRuntimeException("revert Snapshot is not supported");
+        if (canHandle(snapshot,SnapshotOperation.REVERT) == StrategyPriority.CANT_HANDLE) {
```

Comment:
@rafaelweingartner if you consider changing a canHandle() method take care. These are called over all types of resources and should not break contract as to not confuse any Orchestrators in the system.